docs/content/docs/dev/table/materialized-table/overview.md

```diff
@@ -0,0 +1,65 @@
+{{< /hint >}}
+
+# Core Concepts
+
+Materialized tables encompass the following core concepts: Data Freshness, Refresh Mode, Query Definition, and Table Schema.
+
+## Data Freshness
+
+Data freshness measures the lag of the materialized table compared to the upstream table. After creating a materialized table, the flink framework will strive to meet the data freshness requirements, though it cannot guarantee complete compliance.
+
+Data freshness is a crucial attribute of a materialized table, serving two main purposes:
+1. **Determining the Refresh Mode**: Currently, there are CONTINUOUS and FULL modes. For details on how to determine the refresh mode, refer to the [materialized-table.refresh-mode.freshness-threshold]({{< ref "docs/dev/table/config" >}}#materialized-table-refresh-mode-freshness-threshold) configuration item.
+    - CONTINUOUS mode: Real-time jobs continuously refresh the materialized table.
+    - FULL mode: The workflow scheduler periodically triggers batch jobs to refresh the materialized table.
+2. **Determining the Refresh Frequency**:
+    - In CONTINUOUS mode, data freshness is converted into the `checkpoint` frequency of the background real-time job.
+    - In FULL mode, data freshness is converted into the scheduling cycle of the workflow.
+
+## Refresh Mode
+
+There are two refresh modes: FULL and CONTINUOUS. By default, the refresh mode is inferred based on data freshness. For specific business scenarios, users can explicitly specify the refresh mode, which will take precedence over the data freshness inference.
+
+- **CONTINUOUS Mode**: Real-time jobs incrementally update the materialized table data, with the data refresh frequency matching the job's `checkpoint` interval.
+- **FULL Mode**: The scheduler periodically triggers a full update of the materialized table data, with the data refresh cycle matching the job's scheduling cycle.
+    - The default overwrite behavior is at the table level. If there are partition fields and the time partition field format is specified via [partition.fields.#.date-formatter]({{< ref "docs/dev/table/config" >}}#partition-fields-date-formatter), the overwrite is by partition, and only the latest partition is refreshed each time.
+
+## Query Definition
+
+The query definition of a materialized table is similar to a regular query. The query results are used to populate the materialized table. In CONTINUOUS mode, the query results are written to the materialized table in real time, while in FULL mode, the query results overwrite the materialized table.
+
+## Table Schema
+
+The data structure of a materialized table is the same as a regular table. It can declare primary keys, watermarks, and partition fields. The field names and types of the table structure are automatically inferred from the corresponding query, and users cannot specify them.
```

Comment:
The schema definition of a materialized table is the same as the regular table. It can declare primary keys and partition keys. The column names and types of materialized table are automatically inferred from the corresponding query, and users cannot specify them.